### PR TITLE
JT Jones Signoff Head on Sensor Subsystem

### PR DESCRIPTION
**Pull request by JTJones73**

### CODE REVIEW

File: Documentation/Signoffs/Head-On-Sensor.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
+# Head On Position Sensor
+## Functionality
+The purpose of the head on position sensor is to determine the target's location relative to the interceptor. Sensing equipment shall be attached to the front of the interceptor facing the test bed. The sensor will then read in data and make the necessary calculations to determine the X, Y, and Z position of the target. Higher accuracy is expected in the X and Y direction rather than Z, as to get an accurate reading on depth a time of flight sensor would be required with a high sampling rate, and this would be difficult and expensive given the test bed is only 6 feet long [1].
+## Constraints
+* C7: Design a sensor array that can detect approaching objects and relay their locations to the interceptor.
```
Comment (cjdrake42) [inline Documentation/Signoffs/Head-On-Sensor.md]: This constraint is for the overarching system. Make this more applicable to the current signoff or omit the constraint alltogether

Comment (JesseTNRoberts) [reply Documentation/Signoffs/Head-On-Sensor.md]: That's a good point

**Review by JesseTNRoberts — COMMENTED** [review]
This hasn't been updated since my last review. Why was a review requested?

**Comment by JTJones73** [comment]
> This hasn't been updated since my last review. Why was a review requested?

I do not see your review except for saying "That's a good point" in which I have not requested an additional review since you posted that. I did close the older pull request to rename the pull request here: https://github.com/JTJones73/Capstone2024-Team2/pull/14
I do not see any other feedback; am I missing something?

File: Documentation/Signoffs/Head-On-Sensor.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/Head-On-Sensor.md]: This constraint should start with shall (as all should).

File: Documentation/Signoffs/Head-On-Sensor.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  * This subsystem will act in conjunction with other position sensors to be able to effectively track the target in three deminsions.
+* C23: The head on position sensor shall have a maximum latency of 100ms and provide updates at least 10 times per second.
+* C24: The head on position  sensor shall be able to detect the target with an accuracy of at least 2 inches from six feet away.
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/Head-On-Sensor.md]: What does a 2 inch accuracy mean here? Is that in the x direction or y? Or is that distance?

**Review by JesseTNRoberts — COMMENTED** [review]
The ribbon connector is not sufficient the camera and pi need to be represented. And this design seems to assume that the camera's mounting position is not something that needs to be considered here. But the mounting position and field of view are critical and are things you need to address here.

**Review by JesseTNRoberts — COMMENTED** [review]
The accuracy formula seems incorrect. Why does accuracy get better as the actual deviates from the reported? Further, why is z actual alone?

My next question is, how did you come up with this needed accuracy? Because it seems arbitrary.

File: Documentation/Signoffs/Head-On-Sensor.md [hunk 4]
```diff
@@ -5,10 +5,11 @@ The purpose of the head on position sensor is to determine the target's location
...
 * C24: The head on position sensor shall be able to detect the target with an accuracy of at least 2 inches when the target is between six and two feet away.
-* * When the target is between six and two feet away from the interceptor the reported location shall not be more than 2 inches away from the target. More leniency is granted for the depth axis. To confirm the accuracy of this subsystem the following equation will be used.
+* * When the target is between six and two feet away from the interceptor the reported location shall not be more than 2 inches away from the target. There are a total of 30 lines the target can be on arranged by two rows of 15 strings with four inches between each string four feet from the interceptor. A 200% factor of safety is added because if the reported location is between two lines the wrong line may be reported. To confirm the error of this subsystem is within specifications the following equation will be used.
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/Head-On-Sensor.md]: I don't understand what this is conveying.

**Review by JesseTNRoberts — COMMENTED** [review]
It is not clear to me that a total euclidean error of 2 inches is sufficient to make sure this works. How do you know?

**Review by JesseTNRoberts — COMMENTED** [review]
This looks good and I'm close to signing off. The issue is, I'm not convinced that your proposed accuracy is possible and that the accuracy is sufficient to then know the correct string/string setting. How do you know that this will work?

File: Documentation/Signoffs/Head-On-Sensor.md
Comment (micahr234) [inline Documentation/Signoffs/Head-On-Sensor.md:1]: How are you going to detect the Z distance? Can you explain how the aimer will fire at the right time without Z?

I am a little skeptical about your solution. 3px is very low resolution tolerance. Also I am concerned that false positives will be found. If it does not work, you might have to redesign again later. If you are ok with this risk, I will sign.

**Comment by micahr234** [comment]
I do not know if you are waiting on me, but if you are, you need to re-request a review.

File: Documentation/Signoffs/Head-On-Sensor.md
Comment (JTJones73) [reply Documentation/Signoffs/Head-On-Sensor.md:1]: I accept the risk and the team would like to pursue this method.

**Review by micahr234 — APPROVED** [review]

**Pull request merged**